frontends/p4/toP4/toP4.h

```diff
@@ -72,6 +72,7 @@ class ToP4 : public Inspector {
         listTerminators.pop_back();
     }
     bool isSystemFile(cstring file);
+    bool isSystemInternalFile(cstring file);
```

Comment:
What's the definition of a system internal file. Any private header include prefixed with `_internal`?

Reply:
This is the hacky part. What happened is that in the following loop, `program->objects` would contain not only `#include <pna.p4>` that is directly from the source program, but all the transitive `<*.p4>` libraries, like `<_internal/pna/dev/blocks.p4>`. This will lead to compiler errors of duplicate definitions on the `ToP4` generated code.

https://github.com/p4lang/p4c/blob/b27702f146526c5ebc8f04d330a9d0f5014f6205/frontends/p4/toP4/toP4.cpp#L154

My current solution is to define this special `p4include/_internal/` ("system internal") directory, and filter it out in the `ToP4` looping.

Reply:
Also I have to say I don't understand why the `const IR::P4Program *program` passed to `ToP4` has such a structure, and I don't know whether this is expected.

If there isn't a good explanation, maybe I should open a separate issue to further discuss this.